contracts/ZecaleDispatcher.sol

```diff
@@ -46,57 +55,47 @@ contract ZecaleDispatcher
-    //   ..            <nested_inputs_2> (HO)
-    //   ..            <nested_inputs_2> (LO)
-    //   ..            ...
-    //   ..            ...
-    //   ..            <nested_inputs_2> (HO)
-    //   ..            <nested_inputs_2> (LO)
-    //   ..            <result_2>        (HO)
-    //   ..            <result_2>        (LO)
-    //   ..            <nested_inputs_3> (HO)
-    //   ..            <nested_inputs_3> (LO)
-    //   ..            ...
-    //   ..            ...
-    //   ..            <nested_inputs_3> (HO)
-    //   ..            <nested_inputs_3> (LO)
-    //   ..            <result_3>        (HO)
-    //   ..            <result_3>        (LO)
-    //   ..            <nested_inputs_4> (HO)
-    //   ..            <nested_inputs_4> (LO)
-    //   ..            ...
-    //   ..            ...
-    //   ..            <nested_inputs_4> (HO)
-    //   ..            <nested_inputs_4> (LO)
-    //   ..            <result_4>        (HO)
-    //   ..            <result_4>        (LO)
+    //   02            <nested_inputs_1> (HO) ---
+    //   03            <nested_inputs_1> (LO)   |
+    //   ..            ...                      |
+    //   ..            ...                     nested_tx_1
+    //   ..            <nested_inputs_1> (HO)   |
+    //   ..            <nested_inputs_1> (LO)   |
+    //   ..            <result_1>        (HO)   |
+    //   ..            <result_1>        (LO) ---
+    //   ..                     ...
+    //   ..            <nested_inputs_N> (HO) ---
+    //   ..            <nested_inputs_N> (LO)   |
+    //   ..            ...                      |
+    //   ..            ...                     nested_tx_N
+    //   ..            <nested_inputs_N> (HO)   |
+    //   ..            <nested_inputs_N> (LO)   |
+    //   ..            <result_N>        (HO)   |
+    //   ..            <result_N>        (LO) ---
     //
     // `nested_parameters` are the extra parameters required by the application
     // contract (the application contract is responsible for "binding" these to
     // the nested inputs to the nested proofs).
     function process_batch(
         uint256[18] memory batch_proof,
         uint256[] memory inputs,
-        uint256[] memory nested_parameters,
+        bytes[] memory nested_parameters,
         IZecaleApplication target_application) public returns(bool)
     {
-        // Compute expected inputs per batch (TODO: move this to the constructor)
-        uint256 total_inputs =
-            Groth16BW6_761.num_inputs_from_vk_length(_vk.length);
-        uint256 inputs_per_batch = (total_inputs - 1) / batch_size;
+        // TODO: Remove this (and all emit log calls below) once code is more
+        // thoroughly tested.
+
+        // emit log("_total_inputs", _total_inputs);
+        // emit log("_inputs_per_nested_tx", _inputs_per_nested_tx);
+        // for (uint256 i = 0 ; i < inputs.length ; ++i) {
+        //     emit log("i", inputs[i]);
+        // }
```

Comment:
Are these supposed to be added to the code (i.e. uncommented) for the time being? (in light of the TODO message above), or are they rather aimed to be commented/uncommented for debug purposes (hence why they are commented by default)?

Reply:
The intention is that they are uncommented as required.

Reply:
Got it, thanks